File: servers/DoctorService/Models/Report.cs
using System.ComponentModel.DataAnnotations;

namespace DoctorService.Models
{
    public class Report
    {
        [Key]
        public string Id { get; set; } = string.Empty;

        public string? Title { get; set; }
        
        public string? TemplateId { get; set; }
        
        public string? Content { get; set; } // markdown content
        
        public string? ChatSessionId { get; set; }

        public string? PatientEmail { get; set; } // is user's email

        public bool IsSent { get; set; } = false; // if IsSent is true => doctor cannot edit the report

        public DateTime? SentDate { get; set; }
        
        public DateTime? CreatedDate { get; set; }
        
        public DateTime? UpdatedDate { get; set; }  
    }
}

File: web_app/types/report.ts
export interface Report {
    id: string;
    title?: string;
    templateId?: string;
    content?: string; // markdown content
    chatSessionId?: string;
    patientEmail?: string; // is user's email
    isSent: boolean; // if IsSent is true => doctor cannot edit the report
    sentDate?: string;
    createdDate?: string;
    updatedDate?: string;
}

export interface CreateReportRequest {
    title?: string;
    templateId?: string;
    content: string; // markdown content - required
    chatSessionId?: string;
    patientEmail?: string; // is user's email
}

export interface UpdateReportRequest {
    title?: string;
    templateId?: string;
    content?: string; // markdown content
    patientEmail?: string; // is user's email
    isSent?: boolean; // Allow updating sent status
    sentDate?: string; // Allow updating sent date
}
